gravity.sh

```diff
@@ -269,6 +269,10 @@ gravity_DownloadBlocklistFromUrl() {
     else port=80
     fi
     bad_list=$(pihole -q -adlist "${domain}" | head -n1 | awk -F 'Match found in ' '{print $2}')
+    # added check for https://github.com/pi-hole/pi-hole/issues/3196
+    if [ $(echo -n ${bad_list} | wc -m) -eq 0 ]; then
+		  bad_list='a Regular Expression'
```

Comment:
Looks okay, but not sure if an empty `bad_list` is always a regex.

Reply:
I concur. If it has not been found in an blacklist, it could still have been blocked by entries in `/etc/hosts` or any `/etc/dnsmasq.d` file.

@stefanostatuti The beta version of `pihole -q` does also test regex so it is very likely that what you want to achieve is much easier basing on the most recent code instead of `master`.

Reply:
hello, i still using version 4.X PiHole and for now i don't want upgrade in 5.X so i can't test my code in 5.X branch.
Yes, I agree with You and it's better to fix the gravity.sh from 4.x Branch, ill try to fixit

Reply:
There will be no further v4 releases. The https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md is specific in the requirements for PRs to be approved.